blueprints/route/index.js

```diff
@@ -148,21 +149,30 @@ function updateRouter(action, options) {
   }
 }
 
-function findRouter(options) {
+function findRouterPath(options) {
   let routerPathParts = [options.project.root];
-  let root = 'app';
 
   if (options.dummy && options.project.isEmberCLIAddon()) {
-    routerPathParts = routerPathParts.concat(['tests', 'dummy', root, 'router.js']);
+    routerPathParts.push('tests', 'dummy', 'app');
   } else {
-    routerPathParts = routerPathParts.concat([root, 'router.js']);
+    routerPathParts.push('app');
   }
 
-  return routerPathParts;
+  for (const routerFile of ['router.js', 'router.ts']) {
```

Comment:
I guess we can also check for the `isTypeScriptProject` option, but that doesn't necessarily mean there's a `router.ts` file as well.

Reply:
I think you're right, and I think we should definitely do the right thing if we find `router.ts` even in a non-`isTypeScritpProject` setup. One note, though: what should happen if there are *both* of these present? Presumably we should error on that condition, because it is not clear which one we should update. So maybe:

```js
let jsRouterPath = path.join(...routerPathParts, 'router.js');
let tsRouterPath = path.join(...routerPathParts, 'router.ts');

let jsRouterPathExists = fs.existsSync(jsRouterPath);
let tsRouterPathExists = fs.existsSync(tsRouterPath);

if (jsRouterPathExists && tsRouterPathExists) {
  throw new SilentError('Found both a `router.js` and a `router.ts` file. Please make sure your project only has one or the other.')
}

if (jsRouterPathExists) return jsRouterPath;
if (tsRouterPathExists) return tsRouterPath;

// existing error
```